jupyterhub/app.py

```diff
@@ -2129,6 +2127,7 @@ def init_services(self):
             name = spec['name']
             # get/create orm
             orm_service = orm.Service.find(self.db, name=name)
+            allowed_roles = spec.get('allowed_roles', [])
```

Comment:
This isn't necessary, as the trait you added loads it onto the service object automatically.